### ISSUE TITLE
Explore wrapping missing translations in a span with a lang attribute

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#2775]
Currently if we do not have a translation in the current language, we fall back to another language, usually English. This can confuse screen readers, or at best cause some confusing pronunciation. The [preferred solution](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html) (I think?) is to wrap text that is known to be in another language in a tag with the locale specified in a `lang` attribute, e.g. if the page is in Spanish but "Hello" is not translated, we might generate

`<span lang="en">Hello</span>, me llamo Ken-ichi`

I guess that might prevent a screen reader from attempting to pronounce "Hello" as "Hey-yo."

It doesn't look like the I18n gem or I18n.js supports that out of the box, but there are some configurations to explore that handle missing translations. We would definitely want to try to support that in both Ruby and JS.

**Comment by todtb** [comment]
As per RFC 5646 and Best Current Practice
> For markup languages, such as HTML and XML, language information
      can be added to each part of the document identified by the markup
      structure (including the whole document itself).  For example, one
      could write `<span lang="fr">C'est la vie.</span>` inside a German
      document; the German-speaking user could then access a French-German 
      dictionary to find out what the marked section meant.  If
      the user were listening to that document through a speech
      synthesis interface, this formation could be used to signal the
      synthesizer to appropriately apply French text-to-speech
      pronunciation rules to that span of text, instead of applying the
      inappropriate German rules.

so I think the preferred solution stands.

As for integrating with I18n gem, I've seen some examples of monkey patching https://github.com/ruby-i18n/i18n/blob/master/lib/i18n/backend/fallbacks.rb, but as you mentioned @kueda, not much native support. Was wondering what configurations you were looking to explore?

**Comment by kueda** [comment]
> Was wondering what configurations you were looking to explore?

I haven't given it much thought, so I'm open to ideas. Monkey patching fallbacks definitely seems worth exploring. Related, but you might be able to override the `t` helper, force I18n to raise exceptions on missing translations, and catch those in your override. Might be a less invasive approach, though also maybe more prone to failure.

**Comment by todtb** [comment]
I think I want to do a quick inventory of `t` usages to verify that adding HTML to the string will not cause issues or that we handle where appropriate. That said, I'm looking for `t(:xxxx)`, `t('xxxx')`, `t("xxxx")`, `t :xxxx`, `t 'xxxx'`, etc. 

If as an initial step I attempt to normalize this, which direction should we go? That is, what are the style preferences here (parens on method calls, single vs double quotes for non-interpolated strings, and so on)? I got the impression that there are loose style guidelines, but I'm not sure if they are codified anywhere at the moment.

**Comment by kueda** [comment]
> If as an initial step I attempt to normalize this, which direction should we go?

Personally I like using `=t :xxx` in views where that's pretty much the entire line, `t( "xxx" )` if `xxx` has dots in it, and `t( :xxx )` everywhere else. If theres value in using one consistent form, I think `t( "xxx" )` would be best, since it conforms to both our Ruby and Javascript style. @pleary, any opinion on this?

JS styles are codified in [`.eslintrc`](https://github.com/inaturalist/inaturalist/blob/main/.eslintrc) (there's a [React-specific one](https://github.com/inaturalist/inaturalist/blob/main/app/webpack/.eslintrc) too). We haven't made our Ruby style explicit yet, though generally we tend toward symbols whenever possible and prefer double-quoted strings over single-quoted.
